### PR TITLE
[9.x.x] Fix DepthTexture precision, macros in Core.hlsl and make URP shaders use DeclareDepthTexture.hlsl when sampling depth

### PR DESCRIPTION
**Pull request by ellioman**
### Purpose of this PR
In this PR I'm doing three things:

1. Making sure we have high precision on our depth texture in DeclareDepthTexture.hlsl
2. Adding parameters to our "TEXTURE_X" macros in Core.hlsl  
3. Making sure URP shaders are including DeclareDepthTexture.hlsl instead of declaring the depth texture themselves.

---

### Checklist for PR maker
- [x] Have you added a backport label (if needed)? For example, the `need-backport-2019.3` label. After you backport the PR, the label changes to `backported-2019.3`.
- [ ] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR.
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---

### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/universal%252Fshader-fix-depth-and-macros

### CODE REVIEW

**Comment by phi-lira** [comment]
Changes should be safe. Adding @PaulDemeulenaere as this affects VFX Graph.

**Comment by ellioman** [comment]
Added @thomas-zeng for XR

File: com.unity.render-pipelines.universal/ShaderLibrary/DeclareDepthTexture.hlsl [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #define UNITY_DECLARE_DEPTH_TEXTURE_INCLUDED
 #include "Packages/com.unity.render-pipelines.universal/ShaderLibrary/Core.hlsl"
 
-TEXTURE2D_X(_CameraDepthTexture);
+TEXTURE2D_X_FLOAT(_CameraDepthTexture);
```
Comment (PaulDemeulenaere) [inline com.unity.render-pipelines.universal/ShaderLibrary/DeclareDepthTexture.hlsl]: Ok, this file is already specific to Universal and included by `com.unity.visualeffectgraph\Shaders\RenderPipeline\Universal\VFXCommon.hlsl`
So, it's good to me.

Comment (thomas-zeng) [inline com.unity.render-pipelines.universal/ShaderLibrary/DeclareDepthTexture.hlsl]: Do we need to make the same change to particles.hlsl?
https://github.com/Unity-Technologies/ScriptableRenderPipeline/blob/5817c893649058444c1eaa51b08ecb60c93c10e9/com.unity.render-pipelines.universal/ShaderLibrary/Particles.hlsl#L8

Comment (ellioman) [reply com.unity.render-pipelines.universal/ShaderLibrary/DeclareDepthTexture.hlsl]: Yes.
I've made it include the DeclareDepthTexture and also did it in some other shaders. Basically all shaders using the DepthTexture are now simply including DeclareDepthTexture.hlsl.

**Comment by DennisDeRykeUnity** [comment]
"_ABV for SRP repository - fast-2020.1" is green on Yamato.  I'll get started with manual XR testing today.

**Comment by DennisDeRykeUnity** [comment]
Tested configuration:
Unity 2020.2.0a4 (branch: trunk, changeset: 01c6244fbc06)
URP (branch: universal/shader-fix-depth-and-macros, commit: a92d737)
Oculus Rift
XRSDK
Stereo Instancing

Used repro project from Case 1201208 (URP Template scene plus distortion material based on URP/Particles/Unlit shader).  
Bumped up Bloom settings for obvious effect. 
Added Depth-of-Field PostFX and row of cubes, receding in distance.

Also went through XRAcceptanceTests except Input tests. 

No regressions found.  I'll test on one XR mobile device prior to approving.

**Comment by fabien-unity** [comment]
@DennisDeRykeUnity  : Please make sure HDRP still works as this change affect the core SRP package (TEXTURE2D_X originally was created for HDRP and there are hundreds of instances of them in the code).

**Comment by phi-lira** [comment]
@DennisDeRykeUnity is it ok to merge this one?

**Pull request merged**